File: C++/02-String-Basics/7.Find Duplicate Characters.cpp
// Find Duplicate Characters
// Example text = "programming"
// Output = r g m

#include <iostream>
using namespace std;

int main() {

    string text = "programming";

    for (int i = 0; i < text.length(); i++) {

        int count = 0;

        for (int j = i + 1; j < text.length(); j++) {

            if (text[i] == text[j]) {
                count++;
            }

        }

        if (count > 0) {

            bool alreadyPrinted = false;

            for (int k = 0; k < i; k++) {

                if (text[i] == text[k]) {
                    alreadyPrinted = true;
                }

            }

            if (!alreadyPrinted) {
                cout << text[i] << " ";
            }
        }
    }

    return 0;
}

File: Python/2.String-Basics/7.Find Duplicate Characters.py
# Find duplicate Characters
# Example text= "programming"
# output = r g m


def Find(text):
    if not text:
        return False
    duplicate=""
    
    for i in range(len(text)):

        for j in range(i+1,len(text)):

            if text[i] == text[j]:

                duplicate +=text[i]
    
    return duplicate

text ="programming"
print(Find(text))
